pkg/entitlements/customers.go

```diff
@@ -215,7 +215,16 @@ func (sc *StripeClient) GetCustomerByStripeID(ctx context.Context, customerID st
 
 // UpdateCustomer updates a customer in stripe with the provided params and ID
 func (sc *StripeClient) UpdateCustomer(id string, params *stripe.CustomerParams) (*stripe.Customer, error) {
-	return sc.Client.Customers.Update(id, params)
+	if id == "" || params == nil {
+		return nil, ErrCustomerIDRequired
+	}
+
+	cust, err := sc.Client.Customers.Update(id, params)
+	if err != nil {
+		return nil, err
+	}
+
+	return cust, nil
```

Comment:
nothing extra is happening on error so you can simplify this:

```suggestion
	return sc.Client.Customers.Update(id, params)
```

Reply:
not sure what you mean... the request itself returns `(*stripe.Customer, error)` and was actively erroring (hence why I changed it to check for the error); it was previously 404'ing due to missing ID parameter

Reply:
right; I see the line above that checks for the missing parameter. What I'm suggesting is that you don't actually need the `if err` check line here because you don't handle the error path vs. success path differently (e.g. you aren't logging anything if there is an error, and aren't doing anything after that function call if it was successful. It can simply be:


```
func (sc *StripeClient) UpdateCustomer(id string, params *stripe.CustomerParams) (*stripe.Customer, error) {
	if id == "" || params == nil {
		return nil, ErrCustomerIDRequired
	}

	return sc.Client.Customers.Update(id, params)
```